js/index.js

```diff
@@ -2,41 +2,119 @@
+  
+  price = price.innerHTML;
+  // console.log(price)
 
-  //... your code goes here
+  const subTotal = price * quantityValue;
+  // console.log(subTotal)
+
+  const productSubTotal = product.querySelector('.subtotal span');
+
+  productSubTotal.innerHTML = subTotal;
+ 
+  return productSubTotal;
+  
 }
 
 function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
+  // // ITERATION 2
 
-  // ITERATION 2
-  //... your code goes here
+  const products = document.querySelectorAll('.product');
+  // console.log(products)
+  const productsArr = [...products]
+
+  productsArr.forEach(item => updateSubtotal(item));
+  const subtotalUpdate = productsArr
+  // console.log(productsArr)
+  
 
   // ITERATION 3
-  //... your code goes here
+  const totalValue = document.querySelector('#total-value span');
+  let totalPrice = 0
+
+  // productsArr.forEach(sumFunction)
+
+  // function sumFunction (item){
+  //   totalPrice += Number(item.querySelector('.subtotal span').innerHTML)
+  //   // console.log(typeof item.querySelector('.subtotal span').innerHTML)
+  //   console.log(totalPrice)
+  //   totalValue.innerHTML = totalPrice
+  //   console.log(totalValue)
+  // }
+
+  productsArr.forEach((item) => {
+    totalPrice += Number(item.querySelector('.subtotal span').innerHTML)
+    // console.log(typeof item.querySelector('.subtotal span').innerHTML)
+    // console.log(totalPrice)
+    totalValue.innerHTML = totalPrice
+    // console.log(totalValue)
+  })
+  
 }
 
 // ITERATION 4
 
 function removeProduct(event) {
   const target = event.currentTarget;
   console.log('The target in remove is:', target);
-  //... your code goes here
+
+  // const productLine = document.getElementsByClassName('product');
+  const wholeCart = target.parentNode.parentNode.parentNode
+  const productLine = target.parentNode.parentNode
+  console.log(wholeCart)
+  console.log(productLine)
+
+  wholeCart.removeChild(productLine)
 }
 
 // ITERATION 5
 
+const createNewProduct = document.getElementById('create')
+createNewProduct.addEventListener('click', createProduct)
+
 function createProduct() {
-  //... your code goes here
+ 
+  // console.log(' Test: The create product button was clicked')
+  const nameNewProduct = document.getElementById('name-new-product').value
+  const priceNewProduct = document.getElementById('price-new-product').valueAsNumber
+  
+
+  const productLine = document.querySelector('.product');
+  const parentProductLines = productLine.parentNode
+  console.log(productLine)
+  console.log(parentProductLines)
+  const cloneProductLine = productLine.cloneNode(true);
+
+  
+  parentProductLines.appendChild(cloneProductLine); //How can I make the new line appear as the last product??
```

Comment:
The problem is that you are correctly appending the <tr> as last child but then your are changing the wrong innerHTML. After you append the cloneProductLine you are changing the name and the price of the first product. 

You should first create the new element and then append it:
`function createProduct () {`
  `const nameNewProduct = document.getElementById('name-new-product').value`
  `const priceNewProduct = document.getElementById('price-new-product').valueAsNumber`

  `const productLine = document.querySelector('.product')`
  `const parentProductLines = document.getElementById('productsRows')`
 ` const cloneProductLine = productLine.cloneNode(true)`
  `cloneProductLine.querySelector('.name span').innerHTML = nameNewProduct`
 ` cloneProductLine.querySelector('.price span').innerHTML = priceNewProduct`

  `parentProductLines.appendChild(cloneProductLine)`
`}`